src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,228 @@
+      this[method] = this[method].bind(this);
+    });
+
+  }
+
+  /**
+   * Set the default expandedg from the properties
+   */
+  componentWillMount() {
+    this.setState({
+      expanded: this.props.expanded
+    });
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    let expanded = !this.state.expanded;
+    this.setState({expanded});
+
+    // Call onToggle callback
+    this.props.onToggle(expanded);
+  }
+
+  /**
+   * [Custom component] The (show more/less) link
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getDetailLink() {
+    let {message, details} = this.props;
+    let isDetailed = message && details && details.length;
+
+    // Check if we must not show the detail link at all
+    if (!isDetailed) {
+      return null;
+    }
+
+    // Get text
+    let moreLess = 'more';
+    if (this.state.expanded) {
+      moreLess = 'less';
+    }
+
+    // Render
+    return (
+        <span
+          className="collapsible-toggle-label  clickable"
+          onClick={this.toggleExpanded} >
+          (Show {moreLess})
+        </span>
+      );
+
+  }
+
+  /**
+   * [Custom component] The list of errors in the details
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getDetailMessages() {
+    let {details} = this.props;
+
+    return details.map(function (message) {
+      let msg = message.toString();
+
+      return (
+          <li key={msg}>{msg}</li>
+        );
+    });
+  }
+
+  /**
+   * [Custom component] The fixed message part
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getFixedPart() {
+    let {message} = this.props;
+    let isVisible = !!message;
+
+    // If not visible, just exit
+    if (!isVisible) {
+      return null;
+    }
+
+    // Render the fixed part of the message
+    return (
+        <div className="collapsible-fixed">
+          <Icon
+            family="mini"
+            id="yield"
+            size="mini"
+            className="icon-alert icon-margin-right"
+            color="red" />
+          {message}
+          {this.getDetailLink()}
+        </div>
+      );
+  }
+
+  /**
+   * [Custom component] The toggled message part
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getToggledPart() {
+    let {message, details} = this.props;
+    let isDetailed = !!message && details && details.length;
+
+    // If not expanded or detailed, just exit
+    if (!isDetailed || !this.state.expanded) {
+      return null;
+    }
+
+    // Render the toggled part of the message
+    // (Note: The nested div is used to float the contents when needed)
+    return (
+        <div className="collapsible-toggled">
+          <div>
+            <ul>
+            {this.getDetailMessages()}
+            </ul>
+          </div>
+        </div>
+      );
+
+  }
+
+  /**
+   * React.js Render Function
+   *
+   * @returns {React.Component} - The rendered content
+   */
```

Comment:
Personally, I would suggest to go with adding the JSDoc comment to all of the function headers, even though the comments are be pointless. For visual coherence and documentation reasons. But of course if this does not follow the design guidelines please let me know.